packages/react/src/SteppedUpContent.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ReactNode, useEffect, useRef } from 'react';
+
+import { useStepUp, useIsSteppedUp, useIsAuthenticated } from '@frontegg/react-hooks';
+
+export interface SteppedUpProps {
+  maxAge?: number;
+  render?: (isSteppedUp: boolean) => React.ReactNode | null;
+  children?: ReactNode;
+}
+
+/**
+ * Stepped up content component that shows the wrapped content only when the user is stepped up
+ * The component triggers the step up flow if the user is not stepped up
+ * @param maxAge maximum time in second that the login is valid
+ * @param render render function to be called when user is stepped up
+ * @param children to be shown when user is stepped up (only if render not provided)
+ */
+export const SteppedUpContent: React.FC<SteppedUpProps> = ({ maxAge, render, children }) => {
+  const isAuthenticated = useIsAuthenticated();
+  if (!isAuthenticated) return null;
+
+  const isSteppedUp = useIsSteppedUp({ maxAge });
```

Comment:
I think you are calling hooks conditionally here because of the return in line 20

Reply:
fixed